### ISSUE TITLE
DllNotFoundException: gpg in Unity Project

### ISSUE THREAD

**Issue by Aupuma** [playgameservices/play-games-plugin-for-unity#2028]
a friend and I are developing a game in Unity for Android, and we want to include some leaderboards with the Google Play Games Services, but we can't get the login to work.

We are using for the login the script MainGui from the minimal sample project of the Plugin. When we press the authenticate button it keeps saying "authenticating":

`if (!Social.localUser.authenticated)
        {
            // Authenticate
            mWaitingForAuth = true;
            mStatusText = "Authenticating...";
            Social.localUser.Authenticate((bool success) =>
            {
                mWaitingForAuth = false;
                if (success)
                {
                    mStatusText = "Welcome " + Social.localUser.userName;
                }
                else
                {
                    mStatusText = "Authentication failed.";
                }
            });
        }`

Looking in the logcat, there is an error:
`E/Unity: DllNotFoundException: gpg
at (wrapper managed-to-native) GooglePlayGames.Native.Cwrapper.Builder:GameServices_Builder_Construct ()
at GooglePlayGames.Native.PInvoke.GameServicesBuilder.Create () [0x00000] in <filename unknown>:0 
at GooglePlayGames.Native.NativeClient.InitializeGameServices () [0x00000] in <filename unknown>:0 
at GooglePlayGames.Native.NativeClient.Authenticate (System.Action`2 callback, Boolean silent) [0x00000] in <filename unknown>:0 
at GooglePlayGames.PlayGamesPlatform.Authenticate (System.Action`2 callback, Boolean silent) [0x00000] in <filename unknown>:0 
at GooglePlayGames.PlayGamesPlatform.Authenticate (System.Action`1 callback, Boolean silent) [0x00000] in <filename unknown>:0 
at GooglePlayGames.PlayGamesPlatform.Authenticate (System.Action`1 callback) [0x00000] in <filename unknown>:0 
at GooglePlayGames.PlayGamesLocalUser.Authenticate (System.Action`1 callback) [0x00000] in <filename unknown>:0 
at LogInAndroid.OnGUI () [0x00000] in <filename unknown>:0 `

We have already tried the solution posted in https://github.com/playgameservices/play-games-plugin-for-unity/issues/280#issuecomment-64280552 but that does not work anymore as the build_plugin.sh does not exist in the current version. We've tried other answers as well as removed and re-added the plugin to the project, checked the api keys, built from a different computer, etc.

**Comment by kantagara** [comment]
Same issue here...

**Comment by anthonybarouch** [comment]
same here

**Comment by MaratGilyazov** [comment]
Face the issue when building via CloudBuild, ok when build locally

**Comment by bochelord** [comment]
Same issue here. I'm building locally, nothing on CloudBuild.

**Comment by colender** [comment]
Reimport the Plugin for some reason the .aar gets excluded

**Comment by bochelord** [comment]
I solved it simply adding again the .aar file

**Comment by Aupuma** [comment]
@kaixiang1306 I tried that but this is what happens:
![android fail](https://user-images.githubusercontent.com/22346231/33177225-2dbafd26-d062-11e7-888f-29d8a0f395f3.PNG)

`Resolution failed

Failed to fetch the following dependencies:
com.google.games:gpgs-plugin-support:0+@aar


UnityEngine.Debug:LogError(Object)
Google.Logger:Log(String, LogLevel)
GooglePlayServices.PlayServicesResolver:Log(String, LogLevel)
GooglePlayServices.ResolverVer1_1:LogMissingDependenciesError(List`1)
GooglePlayServices.<DoResolutionUnsafe>c__AnonStorey18:<>m__26(List`1)
GooglePlayServices.<GradleResolution>c__AnonStorey14:<>m__1F(Result)
GooglePlayServices.<GradleResolution>c__AnonStorey15:<>m__28()
GooglePlayServices.PlayServicesResolver:PumpUpdateQueue()
UnityEditor.EditorApplication:Internal_CallUpdateFunctions()
` 
@bochelord Where I can find the .aar file and where do I put it?

**Comment by bochelord** [comment]
@Aupuma sorry, but I grabbed it from another project that we were releasing at the same time. For whatever reason, in one project it worked and in the other one, it didn't. Relatively same architecture.
Of course, for the first project that worked, we applied the Android Dependencies.
Maybe you can try on a clean project and then grab the library from there. Not even close to an ideal solution, rather a dirty workaround, but it worked for us

**Comment by sinmaplewing** [comment]
My jar resolver keep removing the Assets/Plugins/Android/gpgs-plugin-support-0.9.42.aar file.
And after removing it and rebuilding it, I got the similar error: Unable to find gpg.

But after I put the Assets/Plugins/Android/gpgs-plugin-support-0.9.42.aar back, the build is all fine with gpgs......

**Comment by AdriaandeJongh** [comment]
I have this error too and haven't found a resolution yet. It's annoying as fuck as the resolver deletes the .aar every time scripts get reloaded.

**Comment by mvmsaikrishna** [comment]
Just re-import the gpg plugin and play services/force-resolve it should work. It is caused due to missing gpgs-*.aar files. In our case, selecting 'use micro mscorlib' in stripping level removed those files.

**Comment by Reizoe** [comment]
Re-importing and force-resolving still led to nothing. I also turned off stripping so that shouldn't be affecting things either. Not to mention I've never had a gpgs-plugin-support-(ver).aar file to begin with.

**Comment by mvmsaikrishna** [comment]
@Reizoe yeah.. re-importing after selecting 'use micro mscorlib' didn't cause any problem

**Comment by Umkerius** [comment]
I have the similiar problem with MoPub and GPGS in one project. I removed android support libs in Plugins/MoPub/libs and reimported GPGS with force resolution. The problem disappeared

**Comment by irina88** [comment]
The solution proposed by s-kris worked for me. Thank you!

**Comment by Bruno2907** [comment]
For me what seemed to have solved this issue was turning off every setting related to auto update and then reverting to my last backup before this happened.
![captura de tela 9](https://user-images.githubusercontent.com/41448627/42962844-ef43f5ee-8b68-11e8-9a35-7b05b379da90.png)

**Comment by kakapro** [comment]
I found that current GPG version not support ARM64 yet, so just uncheck ARM64 in Build Setting -> Other Settings -> Target Architectures to fix it.

**Comment by setebe** [comment]
i tried all of your advices. but none of them worked.
Just go back to old 2017.4 version and you'll be fine

**Comment by chwuchang** [comment]
Updated to 0.9.64 and this bug (Unable to to load DLL 'gpg') still exist.
Not working on: Il2Cpp-ARM64
Working on: Il2Cpp-ArmV7, Il2Cpp-x86, Mono-all

**Comment by SantoshKetaraju** [comment]
Same here Not working on il2Cpp-arm64-v8a,
working on :  il2Cpp -armeabi-v7a

google is going to remove support for 32 bit from 2021, also from august 1st 2019, Apps must have Arm64 support along with 32bit for the app to be visible in play store for 64bit version devices so it is a matter of urgency to solve this please anyone know or happen to find the solution post it :(

**Comment by SantoshKetaraju** [comment]
> 
> 
> Updated to 0.9.64 and this bug (Unable to to load DLL 'gpg') still exist.
> Not working on: Il2Cpp-ARM64
> Working on: Il2Cpp-ArmV7, Il2Cpp-x86, Mono-all

Did you find any solution?

**Comment by SantoshKetaraju** [comment]
> 
> 
> > Updated to 0.9.64 and this bug (Unable to to load DLL 'gpg') still exist.
> > Not working on: Il2Cpp-ARM64
> > Working on: Il2Cpp-ArmV7, Il2Cpp-x86, Mono-all
> 
> Did you find any solution?

Ok so mine solved after i force resolved from play resolver

**Comment by vaibhavgpt-google** [comment]
Closing this issue as part of a cleanup of older tickets. Google Play Games v1 is now deprecated. Please upgrade to version v2.1.0 or newer: https://github.com/playgameservices/play-games-plugin-for-unity/releases/tag/v2.1.0. If this issue persists on the new v2 architecture, please open a new ticket.
